subprojects/core/src/integTest/groovy/org/gradle/api/tasks/options/TaskOptionIntegrationTest.groovy

```diff
@@ -321,4 +321,121 @@ Options
         then:
         result.assertTaskExecuted(':myTask').assertOutputContains('Serial: 4321')
     }
+
+    def "options from interfaces with same method defined in class should use overridden value"() {
```

Comment:
For completeness' sake, it might be worth adding additional tests to reveal the behavior when classes extends super classes possessing options that conflict with their interfaces.

A extends B and A implements I - where B and I have the same option defined.
A extends B and B implements I - where A and I have the same option defined.
A extends B and B implements I - where A and B and I have the same option defined.